src/battle_ai_main.c

```diff
@@ -3875,14 +3877,11 @@ static u32 AI_CalcMoveScore(u32 battlerAtk, u32 battlerDef, u32 move)
             ADJUST_SCORE(1);
         IncreaseStatUpScore(battlerAtk, battlerDef, STAT_CHANGE_DEF, &score);
         break;
-    case EFFECT_FAKE_OUT:
-        if (move == MOVE_FAKE_OUT) // filter out first impression
-        {
-            if (ShouldFakeOut(battlerAtk, battlerDef, move))
-                ADJUST_SCORE(4);
-            else
-                ADJUST_SCORE(-10);
-        }
+    case EFFECT_FIRST_TURN_ONLY:
+        if (gBattleMoves[move].argument == MOVE_FAKE_OUT && ShouldFakeOut(battlerAtk, battlerDef, move))
+            ADJUST_SCORE(4);
```

Comment:
Instead of checking the move argument, wouldn't simply checking the following be enough?
```
HasMoveEffectWithChance(MOVE_EFFECT_FLINCH, 100)```

Reply:
I think you're right in that Fake Out needs special consideration, btw, sorry for not being more helpful. I think its unique combination of priority, effect and additional effects makes it tricky to catch - but I think the above suggestion is the most sustainable, no?

Reply:
> but I think the above suggestion is the most sustainable, no?
ye, it looks like it. I can remove both arguments now.

Reply:
pushed a change